### ISSUE TITLE
show-unit-syndromes.rb - error when printing FLASH_TILE syndrome

### ISSUE THREAD

**Issue by eroen** [DFHack/dfhack#1117]
[show-unit-syndromes.rb](https://github.com/DFHack/scripts/blob/master/show-unit-syndromes.rb#L732-L733) has this code:
```
  when "FLASH_TILE"
    if !showdisplayeffects then return "", Output::DEFAULT end
    name = "Flash"
    color = ce.sym_color >> 8
    tile = ce.sym_color - (color * 256)
    desc = "tile = #{tile}, colour=#{color}, time=#{ce.period}, period=#{ce.time}"
    color = Output::DEFAULT
```

which sadly stopped working properly after [this change](https://github.com/DFHack/df-structures/commit/86f0181ba172de03a73f88789675e6f8cf885a33#diff-5348c5d50fabd4fe1a69a6807a98e2f4L307):
```patch
...
     <class-type type-name='creature_interaction_effect_flash_symbolst'
￼                  inherits-from='creature_interaction_effect'>
-         <int32_t name="sym_color"/>
+         <static-array type-name='uint8_t' name="sym_color" count='4'/>
￼          <int32_t name="period"/>
￼          <int32_t name="time"/>
...
```

resulting in this output:
```
E: NoMethodError: undefined method `>>' for [165, 3, 0, 0]:DFHack::MemHack::StaticArray
 /opt/dfhack/hack/scripts/show-unit-syndromes.rb:732:in `get_effect'
 /opt/dfhack/hack/scripts/show-unit-syndromes.rb:908:in `block (2 levels) in <top (required)>'
 /opt/dfhack/hack/ruby/ruby-autogen-defs.rb:420:in `block in each'
 /opt/dfhack/hack/ruby/ruby-autogen-defs.rb:420:in `each'
 /opt/dfhack/hack/ruby/ruby-autogen-defs.rb:420:in `each'
 /opt/dfhack/hack/scripts/show-unit-syndromes.rb:908:in `collect'
 /opt/dfhack/hack/scripts/show-unit-syndromes.rb:908:in `block in <top (required)>'
 /opt/dfhack/hack/scripts/show-unit-syndromes.rb:1032:in `[]'
```

It should be a trivial fix, but I don't really know ruby or the dfhack codebase. I assume that either `color = ce.sym_color[0]` or `color = ce.sym_color[3]`  would give the intended result though.

**Comment by lethosor** [comment]
Looks like sym_color[1] to me - I'm thinking sym_color is `[tile, fg, bg, bold]`, which would match DF's screen buffer format, and '3' is a valid foreground color (cyan).

**Comment by lethosor** [comment]
@BenLubar you mentioned something broken with show-unit-syndromes recently - was this it?

**Comment by BenLubar** [comment]
@lethosor `show-unit-syndromes world` gives an error like this:

```
E: NoMethodError: undefined method `name' for #<DFHack::CreatureInteractionEffectCanDoInteractionst:0x0001df08817f70>
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/scripts/show-unit-syndromes.rb:315:in `get_interaction'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/scripts/show-unit-syndromes.rb:766:in `get_effect'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/scripts/show-unit-syndromes.rb:908:in `block (2 levels) in <top (required)>'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/ruby/ruby-autogen-defs.rb:420:in `block in each'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/ruby/ruby-autogen-defs.rb:420:in `each'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/ruby/ruby-autogen-defs.rb:420:in `each'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/scripts/show-unit-syndromes.rb:908:in `collect'
 C:/Users/Owner/Documents/GitHub/dfhack/build/win64/DF/hack/scripts/show-unit-syndromes.rb:908:in `block in <top (required)>'
```

**Comment by lethosor** [comment]
Ah, something else. I'll look at that too if you haven't fixed it yet.

**Comment by lethosor** [comment]
@BenLubar fixed yours in dfhack/scripts@3bc50dd
